common/lib/container-definitions/src/loader.ts

```diff
@@ -252,20 +252,27 @@ export interface IContainer extends IEventProvider<IContainerEvents>, IFluidRout
 
     /**
      * Boolean indicating whether the container is currently connected or not
+     * @deprecated - 0.58, This API will be removed in 0.60.0
+     * Use `connectionState.connected` instead
+     * See https://github.com/microsoft/FluidFramework/issues/9167 for context
      */
     readonly connected: boolean;
 
     /**
      * Dictates whether or not the current container will automatically attempt to reconnect to the delta stream
      * after receiving a disconnect event
      * @param reconnect - Boolean indicating if reconnect should automatically occur
-     * @alpha
+     * @deprecated - 0.58, This API will be removed in 0.60.0
+     * Use `connect()` and `disconnect()` instead of `setAutoReconnect(true)` and `setAutoReconnect(false)` respectively
+     * See https://github.com/microsoft/FluidFramework/issues/9167 for context
      */
     setAutoReconnect?(reconnect: boolean): void;
 
     /**
      * Have the container attempt to resume processing ops
-     * @alpha
+     * @deprecated - 0.58, This API will be removed in 0.60.0
+     * Use `connect()` instead
```

Comment:
Should we add `connect()` and `disconnect()` as optional right away, especially since you're mentioning them here, and to help consumers transition?  I don't think adding an optional thing is ever breaking (but could be wrong).

Reply:
I actually had them added as optional earlier but changed it after discussion [here](https://github.com/microsoft/FluidFramework/pull/9439#discussion_r824881606). To summarize, I will have to add the functions to `IFluidContainer` in a separate PR anyway, so I am planning to add them to `IContainer` and `IFluidContainer` in one PR. This way we won't have to make another PR to change it from optional to mandatory functions.

Reply:
Ok.  Feel free to ignore what I'm about to say then :)

In general, I believe having a property absent is completely equivalent to having it present but optional, in terms of compatibility.  It's just a hint saying "you may find something here...".  So I would think you can and should add the optional functions to both interfaces in this PR.

> This way we won't have to make another PR to change it from optional to mandatory

To say the above another way, changing a member from optional to mandatory is equivalent to adding it when it's missing.  At least I think so - correct me if I'm wrong, this stuff really bends my brain.

Reply:
@markfields The main problem with going from optional to mandatory I think is that it can still be a breaking change to anyone with an external implementation of the interface, where an implementation/object matching the interface suddenly no longer does so.  It sounds like this shouldn't be an issue in this case though, but if there's a breaking change no matter how you slice it there's some appeal to doing just one step instead of two

Reply:
I'm saying something different, I agree with that for sure.

Going from nothing to optional is never any trouble. Going from optional to required is trouble - it's as much trouble as adding a required thing outright. So might as well add optional earlier than later to give people an easier time onboarding/preparing.

Reply:
What's the appeal of doing one step instead of two?

Reply:
I agree with @heliocliu  and I'm not sure why this was removed from this PR? The goal of this PR is to introduce connect and disconnect at the Container level. Since we are enforcing using IContainer we are telling devs to use APIs as part of depreciation that aren't actually exposed.

Seems like the right step is to add them as optional as part of this PR and include the version in the Breaking.md that we will be making the breaking change. Then make them mandatory in next (or whatever breaking change we decide)

Reply:
> What's the appeal of doing one step instead of two?

Fewer steps 🙃